sdk/core/System.ClientModel/src/Internal/SSE/ServerSentEvent.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace System.ClientModel.Internal;
+
+/// <summary>
+/// Represents an SSE event.
+/// See SSE specification: https://html.spec.whatwg.org/multipage/server-sent-events.html
+/// </summary>
+internal readonly struct ServerSentEvent
+{
+    // Gets the value of the SSE "event type" buffer, used to distinguish between event kinds.
+    public string EventType { get; }
+
+    // Gets the value of the SSE "data" buffer, which holds the payload of the server-sent event.
+    public string Data { get; }
+
+    // Gets the value of the "last event ID" buffer, with which a user agent can reestablish a session.
+    public string? Id { get; }
+
+    // If present, gets the defined "retry" value for the event, which represents the delay before reconnecting.
+    public TimeSpan? ReconnectionTime { get; }
+
+    public ServerSentEvent(string type, string data, string? id, string? retry)
+    {
+        EventType = type;
+        Data = data;
+        Id = id;
+        ReconnectionTime = retry is null ? null :
+            int.TryParse(retry, out int time) ? TimeSpan.FromMilliseconds(time) : null;
```

Comment:
```suggestion
        ReconnectionTime = retry switch
        {
            null => null,
            _ when int.TryParse(retry, out int time) =>  TimeSpan.FromMilliseconds(time)
            _ => null
        };
```

nit:  It may be more readable to avoid the nested ternaries.

Reply:
Addressed.